### PR TITLE
Bring our branch up to date with upstream

### PR DESCRIPTION
**Pull request by ukd1**

### CODE REVIEW

File: lib/queue_classic/tasks.rb [hunk 1]
```diff
@@ -11,11 +13,8 @@
     @worker = QC.default_worker_class.new
 
     trap('INT') do
-      $stderr.puts("Received INT. Shutting down.")
-      if !@worker.running
-        $stderr.puts("Worker has stopped running. Exit.")
-        exit(1)
-      end
+      $stderr.puts("Received INT. Shutting down.")      
+      abort("Worker has stopped running. Exit.") unless @worker.running
```
Comment (magni-) [inline lib/queue_classic/tasks.rb:17]: We used to also `exit(1)` in this case, but no longer do.

File: lib/queue_classic/worker.rb [hunk 2]
```diff
@@ -112,7 +114,10 @@ def process(queue, job)
           queue.delete(job[:id])
           finished = true
         end
-      rescue => e
+      rescue StandardError, ScriptError, NoMemoryError => e
```
Comment (magni-) [inline lib/queue_classic/worker.rb:117]: This is the only other possible surprise I can see, if there were some other error we were silently handling here up til now.

**Review by magni- — APPROVED** [review]
Commented on the only two non-trivial changes (apart from the obvious skip locked change)

File: lib/queue_classic/tasks.rb (hunk 1 above)
Comment (magni-) [reply lib/queue_classic/tasks.rb:17]: Oh wait I missed that the `abort` is new, we were doing the same thing in two steps above. 🙈

File: lib/queue_classic/worker.rb (hunk 2 above)
Comment (ukd1) [reply lib/queue_classic/worker.rb:117]: yep, but, prolly we shouldn't be right? but might cause a few new unseen errors?

Comment (magni-) [reply lib/queue_classic/worker.rb:117]: Right, I don't expect there to be any, and this isn't a blocker to upgrading... just something to be aware of I guess? If we see new errors in Sentry, they're probably from here?

**Comment by ukd1** [comment]
@shosti can you once over this (@magni- already has) and lmk if there is anything else. This is the full diff between what we're using in prod and QC master.

**Comment by ukd1** [comment]
@shosti merging, as we've merged.

**Pull request merged**